docs/tags.rst

```diff
@@ -0,0 +1,38 @@
+Manage Tags
+===========
+
+Lance, much like Git, employs the :py:attr:`LanceDataset.tags <lance.LanceDataset.tags>`
+property to label specific versions within a dataset's history.
+
+:py:class:`Tags <lance.dataset.Tags>` are particularly useful for tracking the evolution of datasets,
+especially in machine learning workflows where datasets are frequently updated.
+For example, you can :py:meth:`create <lance.dataset.Tags.create>`, :meth:`update <lance.dataset.Tags.update>`,
+and :meth:`delete <lance.dataset.Tags.delete>` or :py:meth:`list <lance.dataset.Tags.list>` tags.
+
+
```

Comment:
We should mention that adding a tag makes it exempt from the cleanup process. I think users can view this as a feature: if they want a particular version preserved, they can tag it. And if they later want to delete that version, they should delete the tag, and then run cleanup.

Reply:
It is mentioned https://github.com/lancedb/lance/pull/3562/files#diff-3c0e591b6a64801b9e3f92380ab99c1546e03faaa769b6de4b37af58d95c588cR286. 

I can modify the wording for delete tag